helix-core/src/main/java/org/apache/helix/util/RebalanceUtil.java

```diff
@@ -150,17 +151,25 @@
    *
    * @param clusterName target cluster name
    * @param enabled enable/disable controller to management mode pipeline
+   * @param clusterStatusMonitor
    */
-  public static void enableManagementMode(String clusterName, boolean enabled) {
+  public static void enableManagementMode(String clusterName, boolean enabled,
+      ClusterStatusMonitor clusterStatusMonitor) {
     GenericHelixController leaderController =
         GenericHelixController.getLeaderController(clusterName);
     if (leaderController != null) {
       LOG.info("Switching management mode pipeline for cluster={}, enabled={}", clusterName,
           enabled);
       leaderController.setInManagementMode(enabled);
     } else {
-      LOG.error("Failed to switch management mode pipeline, enabled={}. "
-          + "Controller for cluster {} does not exist", clusterName, enabled);
+      throw new HelixException(String.format("Failed to switch management mode pipeline, "
+          + "enabled=%s. Controller for cluster %s does not exist", enabled, clusterName));
+    }
+
+    // Cluster is paused/frozen.
+    if (clusterStatusMonitor != null) {
+      clusterStatusMonitor.setEnabled(!enabled);
+      clusterStatusMonitor.setPaused(enabled);
```

Comment:
As mentioned above, the monitor is an internal field of the GenericHelixController. We should keep the operations inside the controller class as long as it is possible. Otherwise, this util could be misused and cause some unexpected side effects.